File: sqry-core/fuzz/corpus/r_plugin/seeds/comprehensive.R
library(stats)

add <- function(a, b) {
  return(a + b)
}

multiply <- function(a, b) {
  a * b
}

Calculator <- R6::R6Class("Calculator",
  public = list(
    value = 0,
    initialize = function() {
      self$value <- 0
    },
    add = function(x) {
      self$value <- self$value + x
      invisible(self)
    }
  )
)

setClass("Point",
  slots = c(x = "numeric", y = "numeric")
)

File: sqry-core/fuzz/corpus/ruby_plugin/seeds/comprehensive.rb
module Math
  class Calculator
    attr_accessor :value

    def initialize
      @value = 0
    end

    def add(a, b)
      a + b
    end

    def self.multiply(a, b)
      a * b
    end

    private

    def private_method
      "private"
    end
  end
end

class Point < Struct.new(:x, :y)
  def distance_from_origin
    Math.sqrt(x**2 + y**2)
  end
end
